src/components/layout/UserPortalUpdatePrompts.js

```diff
@@ -0,0 +1,165 @@
+
+    const onUpdateNow = () => {
+        const updateUrl = userPortalResp?.update_url;
+        if (updateUrl) {
+            const portalUrl = `${updateUrl}&redirectUrl=${encodeURIComponent(
+                window.location.href
+            )}`;
+            window.location.replace(portalUrl);
+        }
+    };
+
+    const onLearnMore = () => {
+        window.open(constants.USER_PORTAL_FAQ, "_blank");
+    };
+
+    return (
+        <>
+            <Snackbar
+                id={ids.USER_PORTAL_ANNOUNCER}
+                anchorOrigin={{
+                    vertical: "top",
+                    horizontal: "center",
+                }}
+                open={showSnackbar}
+            >
+                <Alert
+                    variant="filled"
+                    severity="warning"
+                    style={{
+                        color: theme.palette.warning.contrastText,
+                    }}
+                    action={[
+                        <Button
+                            variant="outlined"
+                            onClick={onLearnMore}
+                            id={build(
+                                ids.USER_PORTAL_ANNOUNCER,
+                                ids.LEARN_MORE
+                            )}
+                            key={ids.LEARN_MORE}
+                            classes={{ root: classes.announcerBtn }}
+                        >
+                            <Typography>{t("learnMore")}</Typography>
+                        </Button>,
+                        <Button
+                            variant="outlined"
+                            onClick={onUpdateNow}
+                            id={build(
+                                ids.USER_PORTAL_ANNOUNCER,
+                                ids.UPDATE_BTN
+                            )}
+                            key={ids.UPDATE_BTN}
+                        >
+                            <Typography>{t("updateNow")}</Typography>
+                        </Button>,
+                        <IconButton
+                            id={build(ids.USER_PORTAL_ANNOUNCER, ids.CLOSE_BTN)}
+                            key={ids.CLOSE_BTN}
+                            size="small"
+                            style={{
+                                color: theme.palette.warning.contrastText,
+                            }}
+                            onClick={onCloseAnnouncer}
+                        >
+                            <Close />
+                        </IconButton>,
+                    ]}
+                >
+                    <AlertTitle>{t("updateAccountInformation")}</AlertTitle>
+                    {t("userPortalGracePeriodText")}
+                </Alert>
+            </Snackbar>
+
+            <Dialog
+                open={!!userPortalResp?.update_required}
+                id={baseId}
+                fullWidth
+                aria-labelledby={dialogTitleId}
+            >
```

Comment:
Should this Dialog set the `disableBackdropClick` and `disableEscapeKeyDown` props to prevent the user from closing the dialog with those actions?
```suggestion
            <Dialog
                open={!!userPortalResp?.update_required}
                id={baseId}
                fullWidth
                disableBackdropClick
                disableEscapeKeyDown
                aria-labelledby={dialogTitleId}
            >
```

Reply:
Ah good catch.  I tried this locally and, at least in FireFox, was unable to close the dialog by clicking outside or hitting Escape, but I'll add these to be safe.

Reply:
Oh, maybe those only matter if `onClose` is also set, but it's not set here.
Oh well, doesn't hurt to add them anyway 🤷